### ISSUE TITLE
LET with in()/out() always returns null results

### ISSUE THREAD

**Issue by obi1kenobi** [orientechnologies/orientdb#7754]
## Expected behavior  
The following two queries should produce identical results:
```
SELECT $bar.quux AS quux FROM (
    MATCH {
        class: Foo,
        as: Foo___1
    }
    RETURN $matches
)
LET $bar = Foo___1.out('Foo_Bar')
```
and
```
SELECT Foo___1.out('Foo_Bar').quux AS quux FROM (
    MATCH {
        class: Foo,
        as: Foo___1
    }
    RETURN $matches
)
```

## Actual behavior  
In the query with the `LET` clause, the value of `quux` is `null` in every row instead of being the same as in the other query.

## Steps to reproduce  
E.g. use the GamesOfThrones prebuilt dataset, and try the following query:
```
SELECT $bar.name AS names
(
    MATCH {
        class: Noblehouse,
        as: Noblehouse___1
    }
    RETURN $matches
)
LET $bar = Noblehouse___1.in('Has_Allegiance')
```

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

I just pushed a fix on 2.2.x, this should solve the problem.
Leaving open only to port the fix to 3.0 on the legacy executor

Thanks

Luigi

**Comment by obi1kenobi** [comment]
Thanks for the quick fix, looking forward to the next release!

**Comment by luigidellaquila** [comment]
Fix ported to develop branch (3.0).
Closing, thanks!

Luigi
